src/open_samus_returns_rando/specific_patches/metroid_patches.py

```diff
@@ -53,259 +54,95 @@ def _patch_metroids(editor: PatcherEditor):
+        "ST_SG_Alpha_003B", [-18800.0, -6100.0, 0.0], 90, "ST_SG_Alpha_003B_Out", ["collision_camera_028"]
+    ),
+    NewStartPoint(
+        "s025_area2b", "TG_SetCheckpoint_002_Gamma_001", [-1800.0, -4500.0, 0.0],
+        "ST_SG_Gamma_001B", [-1550.0, -4500.0, 0.0], -90, "ST_SG_Gamma_001B_Out", ["collision_camera039"]
+    ),
+    NewStartPoint(
+        "s060_area6", "TG_SetCheckpoint_002_Alpha_001", [3100.0, -6100.0, 0.0],
+        "ST_SG_Alpha_001B", [2850.0, -6100.0, 0.0], 90, "ST_SG_Alpha_001B_Out", ["collision_camera_015"]
+    ),
+    NewStartPoint(
+        "s070_area7", "TG_SetCheckpoint_002_Zeta_001", [6600.0, 7500.0, 0.0],
+        "ST_SG_Zeta_001B", [6850.0, 7500.0, 0.0], -90, "ST_SG_Zeta_001B_Out", ["collision_camera_040"]
+    ),
+    NewStartPoint(
+        "s070_area7", None, None,
+        None, [2400.0, 7600.0, 0.0], 90, "ST_SG_Omega_001_Out", ["collision_camera_047"]
+    ),
+    NewStartPoint(
+        "s070_area7", "TG_SetCheckpoint_002_Omega_001", [-2000.0, 7500.0, 0.0],
+        "ST_SG_Omega_001B", [-2250.0, 7500.0, 0.0], 90, "ST_SG_Omega_001B_Out", ["collision_camera_046"]
+    ),
+    NewStartPoint(
+        "s070_area7", "TG_SetCheckpoint_003_Omega_001", [-1025.0, 9700.0, 0.0],
+        "ST_SG_Omega_001C", [-1025.0, 9700.0, 0.0], -90, None, ["collision_camera_061"]
+    ),
+]
+
+
+def add_startpoints(editor: PatcherEditor, new_startpoint: NewStartPoint):
+    template_tg = editor.get_scenario("s025_area2b").raw.actors[0]["TG_SetCheckpoint_001_Gamma_001"]
+    template_st = editor.get_scenario("s025_area2b").raw.actors[5]["ST_SG_Gamma_001"]
+
+    scenario_name = new_startpoint.scenario
+    scenario_file = editor.get_scenario(scenario_name)
+
+    # Copy the actors
+    if new_startpoint.tg_name is not None:
+        editor.copy_actor(scenario_name, new_startpoint.tg_position, template_tg, new_startpoint.tg_name, 0)
+    if new_startpoint.st_name is not None:
+        editor.copy_actor(scenario_name, new_startpoint.st_position, template_st, new_startpoint.st_name, 5)
+    if new_startpoint.st_out_name is not None:
+        editor.copy_actor(scenario_name, new_startpoint.st_position, template_st, new_startpoint.st_out_name, 5)
+
+    # Add to the entity groups
+    for entity_group in new_startpoint.entity_groups:
+        for actor in [new_startpoint.tg_name, new_startpoint.st_name, new_startpoint.st_out_name]:
+            if actor is not None:
+                scenario_file.add_actor_to_entity_groups(entity_group, actor, True)
+
+    # Rotate the startpoint actors
+    if new_startpoint.st_name is not None:
+        scenario_file.raw.actors[5][new_startpoint.st_name]["rotation"][1] = new_startpoint.st_rotation
+        if new_startpoint.st_out_name is not None:
+            scenario_file.raw.actors[5][new_startpoint.st_out_name]["rotation"][1] = (
+            -abs(new_startpoint.st_rotation)
+            if new_startpoint.st_rotation == 90
+            else abs(new_startpoint.st_rotation)
+        )
```

Comment:
if the value is 90 it does `-abs(new_startpoint.st_rotation)` but this is always just `-90`.
The whole ternary operator is just doing `-1 * new_startpoint.st_rotation`, isn't it?

Reply:
Yes it's always the opposite of the standard actor.

Reply:
I guess I could just hardcode the out actor's rotation based on the rotation value given. Would be simpler.

Reply:
Don't know what you had against `-1 * new_startpoint.st_rotation` but whatever.